src/main/java/seedu/address/model/person/Company.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an Internship's company name in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ */
+public class Company {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Company names should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String fullName;
+
+    /**
+     * Constructs a {@code CompanyName}.
+     *
+     * @param name A valid company name.
+     */
+    public Company(String name) {
+        if (name != null) {
+            checkArgument(isValidName(name), MESSAGE_CONSTRAINTS);
+            fullName = name;
+        } else {
+            fullName = null;
+        }
+    }
+
+    public static boolean isValidName(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return fullName;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof seedu.address.model.person.Company // instanceof handles nulls
+                && fullName.equals(((seedu.address.model.person.Company) other).fullName)); // state check
```

Comment:
If fullName is null then it causes NullPointerException from `equals()`.
So we need enhanced checks.
```suggestion
        if (other == this) {
            // short circuit if same object
            return true;
        } else if (other instanceof Company) {
            // instanceof handles nulls
            if (fullName == null || ((Company) other).fullName == null) {
                // at least 1 of the values are null
                return fullName == ((Company) other).fullName;
            }
            return fullName.equals(((Company) other).fullName); // state check
        }
        return false; // this is not null while other is null
```